jenkins-build-pack-api/Jenkinsfile

```diff
@@ -883,102 +582,108 @@ def updateServiceWithEndpointInfo(service, domain, serviceUrl, endpointKey, endp
-	body = JsonOutput.toJson([
-					from : fromStr,
-					to : email_id,
-					subject : body_subject,
-					text : body_text,
-					cc : cc_email,
-					html : body_html
-			    ])
-
-   	try{
-		def sendMail = sh (script: "curl -X POST \
-				${g_base_url}/platform/email \
-				-k -v -H \"Authorization: $g_login_token\" \
-				-H \"Content-Type: application/json\" \
-				-d \'${body}\'", returnStdout: true).trim()
-		def responseJSON = parseJson(sendMail)
-		if(responseJSON.data){
-			echo "successfully sent e-mail to $email_id"
+def send_status_email (config, build_status, email_content) {
+	if(domain != "platform"){
+		echo "Sending build notification to ${config['created_by']}"
+		def body_subject = ''
+		def body_text = ''
+		def cc_email = ''
+		def body_html = ''
+		if (build_status == 'STARTED') {
+			echo "email status started"
+			body_subject = 'Jazz Build Notification: Deployment STARTED for service: ' + config['service']
+		} else if (build_status == 'FAILED') {
+			echo "email status failed"
+			def build_url = env.BUILD_URL + 'console'
+			body_subject = 'Jazz Build Notification: Deployment FAILED for service: ' + config['service']
+			body_text = body_text + '\n\nFor more details, please click this link: ' + build_url
+		} else if (build_status == 'COMPLETED'){
+			body_subject = 'Jazz Build Notification: Deployment COMPLETED successfully for service: ' + config['service']
 		} else {
-			echo "exception occured while sending e-mail: $responseJSON"
+			echo "Unsupported build status, nothing to email.."
+			return
+		}
+		if (email_content != ''){
+			body_text = body_text + '\n\n' + email_content
 		}
-   	} catch (e){
-       echo "Failed while sending build status notification"
-   	}
+		def fromStr = 'Jazz Admin <' + configLoader.JAZZ.ADMIN + '>'
+		body = JsonOutput.toJson([
+						from : fromStr,
+						to : config['created_by'],
+						subject : body_subject,
+						text : body_text,
+						cc : cc_email,
+						html : body_html
+					])
+
+		try{
+			def sendMail = sh (script: "curl -X POST \
+					${g_base_url}/platform/email \
+					-k -v -H \"Authorization: $auth_token\" \
+					-H \"Content-Type: application/json\" \
+					-d \'${body}\'", returnStdout: true).trim()
+			def responseJSON = parseJson(sendMail)
+			if(responseJSON.data){
+				echo "successfully sent e-mail to ${config['created_by']}"
+			} else {
+				echo "exception occured while sending e-mail: $responseJSON"
+			}
+		} catch (e){
+		   echo "Failed while sending build status notification"
+		}
+	}
 }
-
-def updateServiceInDB(service, domain, serviceUrl,status) {
+def updateServiceInDB(config,status) {
 
 	def updateJson = ''
-
+	def serviceUrl = g_base_url + '/platform/services'
 	if(status) {
 		updateJson = "{ \"status\": \"$status\" }"
 
 		echo "new json $updateJson"
 
-		def serviceUpdateUrl = "$serviceUrl/$service_id?isAdmin=true"
+		def serviceUpdateUrl = "$serviceUrl/${config['service_id']}?isAdmin=true"
```

Comment:
good to see this. When involved in future code reviews in OSS where you are looking at other peoples code, please make sure this consistency is maintained.